### PR TITLE
python3Packages.httpx: 0.16.1 -> 0.17.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 0.17.0

Change log: https://github.com/encode/httpx/releases/tag/0.17.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 114703` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>28 packages built:</summary>
  <ul>
    <li>datasette (python38Packages.datasette)</li>
    <li>google-music-scripts</li>
    <li>home-assistant</li>
    <li>python37Packages.PyRMVtransport</li>
    <li>python37Packages.axis</li>
    <li>python37Packages.datasette</li>
    <li>python37Packages.elmax</li>
    <li>python37Packages.entrance</li>
    <li>python37Packages.entrance-with-router-features</li>
    <li>python37Packages.fastapi</li>
    <li>python37Packages.google-music</li>
    <li>python37Packages.httpx</li>
    <li>python37Packages.pytest-httpx</li>
    <li>python37Packages.respx</li>
    <li>python37Packages.sanic</li>
    <li>python37Packages.sanic-auth</li>
    <li>python38Packages.PyRMVtransport</li>
    <li>python38Packages.axis</li>
    <li>python38Packages.elmax</li>
    <li>python38Packages.entrance</li>
    <li>python38Packages.entrance-with-router-features</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.google-music</li>
    <li>python38Packages.httpx</li>
    <li>python38Packages.pytest-httpx</li>
    <li>python38Packages.respx</li>
    <li>python38Packages.sanic</li>
    <li>python38Packages.sanic-auth</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114703` at 25ce56b1 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/e0b7f9228abb542ccd22ccb69a898a60">home-assistant</a></li></li>
  </ul>
</details>
<details>
  <summary>27 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/aedf01a740b6bc5819dcad962ffab377">datasette</a></li> (python38Packages.datasette)</li>
    <li><a href="https://gist.github.com/b0a82061f3209d1e05103fe877d579ef">google-music-scripts</a></li></li>
    <li><a href="https://gist.github.com/6c1bbb4300185b5a574badadc5013e00">python37Packages.PyRMVtransport</a></li></li>
    <li><a href="https://gist.github.com/c033875fc60ea080813d827989a093a1">python37Packages.axis</a></li></li>
    <li><a href="https://gist.github.com/153ce699639d5abf69a04d19ff82e050">python37Packages.datasette</a></li></li>
    <li><a href="https://gist.github.com/7d0b4bdf5268208787a5c1ba239df2d4">python37Packages.elmax</a></li></li>
    <li><a href="https://gist.github.com/04d44b6caa652f70497c6587e0f59e52">python37Packages.entrance</a></li></li>
    <li><a href="https://gist.github.com/9660722d3ed50cee9373ffafb6a6836c">python37Packages.entrance-with-router-features</a></li></li>
    <li><a href="https://gist.github.com/9e09087aa8a70a7e4d961504540d051f">python37Packages.fastapi</a></li></li>
    <li><a href="https://gist.github.com/8fbfcd4335bddb816a67788f8d9a8022">python37Packages.google-music</a></li></li>
    <li><a href="https://gist.github.com/93917ed21277d94fed5c8e42894d3ecd">python37Packages.httpx</a></li></li>
    <li><a href="https://gist.github.com/a621187aee7faeb465c7e9ae13201158">python37Packages.pytest-httpx</a></li></li>
    <li><a href="https://gist.github.com/b1a31d6231e882d8c9c3f3893c6b76bc">python37Packages.respx</a></li></li>
    <li><a href="https://gist.github.com/a188ef68ff5ec21a8363a979aed8d3ff">python37Packages.sanic</a></li></li>
    <li><a href="https://gist.github.com/cd8d87a36930731d181f9a9e5d414ea4">python37Packages.sanic-auth</a></li></li>
    <li><a href="https://gist.github.com/42143d9b86995598e3ad12f5d3b3a51f">python38Packages.PyRMVtransport</a></li></li>
    <li><a href="https://gist.github.com/386e6d397b9b4050ed1b585fd2778e89">python38Packages.axis</a></li></li>
    <li><a href="https://gist.github.com/8cd4b432b1003dc9972eeeaa34a3b973">python38Packages.elmax</a></li></li>
    <li><a href="https://gist.github.com/802767bf5c8859001822d4a04e859946">python38Packages.entrance</a></li></li>
    <li><a href="https://gist.github.com/c949a022a5d712d85724c3c6c5d6c051">python38Packages.entrance-with-router-features</a></li></li>
    <li><a href="https://gist.github.com/1ecb8d4456e58e47cac8dcde3a1f5196">python38Packages.fastapi</a></li></li>
    <li><a href="https://gist.github.com/2f3a58ffd711e31f0a3dd10a32a46341">python38Packages.google-music</a></li></li>
    <li><a href="https://gist.github.com/22c90939115694e07c1e2f6a3f3461ea">python38Packages.httpx</a></li></li>
    <li><a href="https://gist.github.com/7f7de2608994319bb0d5d2640505dd4b">python38Packages.pytest-httpx</a></li></li>
    <li><a href="https://gist.github.com/fd60801845a4c259db7e10b04e4f0fa8">python38Packages.respx</a></li></li>
    <li><a href="https://gist.github.com/eba844382a1501e58376be94828ae3c2">python38Packages.sanic</a></li></li>
    <li><a href="https://gist.github.com/20905a4ec9dcc90bc898d8dddf156467">python38Packages.sanic-auth</a></li></li>
  </ul>
</details>

File: pkgs/development/python-modules/pytest-httpx/default.nix [hunk 1]
```diff
@@ -13,6 +13,11 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [ httpx pytest ];
 
+  postPatch = ''
+    # https://github.com/Colin-b/pytest_httpx/issues/35
+    substituteInPlace setup.py --replace "httpx==0.16.*" "httpx>=0.16.*"
```
Comment (dotlambda) [inline pkgs/development/python-modules/pytest-httpx/default.nix:18]: You should rather update to https://github.com/Colin-b/pytest_httpx/releases/tag/v0.11.0.

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pytest-httpx/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/pytest-httpx/default.nix]: Oh, nice that they already providing an update.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 114703` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>28 packages built:</summary>
  <ul>
    <li>datasette (python38Packages.datasette)</li>
    <li>google-music-scripts</li>
    <li>home-assistant</li>
    <li>python37Packages.PyRMVtransport</li>
    <li>python37Packages.axis</li>
    <li>python37Packages.datasette</li>
    <li>python37Packages.elmax</li>
    <li>python37Packages.entrance</li>
    <li>python37Packages.entrance-with-router-features</li>
    <li>python37Packages.fastapi</li>
    <li>python37Packages.google-music</li>
    <li>python37Packages.httpx</li>
    <li>python37Packages.pytest-httpx</li>
    <li>python37Packages.respx</li>
    <li>python37Packages.sanic</li>
    <li>python37Packages.sanic-auth</li>
    <li>python38Packages.PyRMVtransport</li>
    <li>python38Packages.axis</li>
    <li>python38Packages.elmax</li>
    <li>python38Packages.entrance</li>
    <li>python38Packages.entrance-with-router-features</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.google-music</li>
    <li>python38Packages.httpx</li>
    <li>python38Packages.pytest-httpx</li>
    <li>python38Packages.respx</li>
    <li>python38Packages.sanic</li>
    <li>python38Packages.sanic-auth</li>
  </ul>
</details>

**Pull request merged**